src/api/app/controllers/source_project_package_meta_controller.rb

```diff
@@ -0,0 +1,95 @@
+class SourceProjectPackageMetaController < SourceController
+  # override the ApplicationController version
+  # to have meaningful error messages
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
+  before_action :require_package_name, only: [:show, :update]
+  before_action :set_rdata, only: [:update]
```

Comment:
@vpereira 

> the method says exactly what it's doing. set rdata. Technically it's not a filter, since it's not filtering anything and after more refactoring in others source*_controllers it will probably end-up being moved to the parent class, as ```before_action``` where it can be called before others similar actions. If you have any suggestion for a better name, just say it.

I am not only speaking about the name, the fact that you need to define an instance variable that is only use in the method itself already looks like this shouldn't be a filter IMO.

Reply:
It's not a filter. It's a method executed before other actions. if you open any other source* controller you will see the same pattern and therefore this method will be moved up to the parent controller and will be used in all other source_* update actions where the same pattern is repeated over and over again.

Reply:
@vpereira 

> It's not a filter.

I meant that it shouldn't be in a before action not a filter, sorry. 

> if you open any other source* controller you will see the same pattern and therefore this method will be moved up to the parent controller and will be used in all other source_* update actions where the same pattern is repeated over and over again.

Any of them there is needed an instance variable... there is also not difference between writing `set_rdata` and `Xmlhash.parse(request.raw_post)`. I am it is not a big code and it is self explanatory in my opinion more than `set_rdata`.